src/proposal.mligo

```diff
@@ -391,22 +399,22 @@ let flush(n, config, store : nat * config * storage): return =
 
 // Removes an accepted and finished proposal by key.
 let drop_proposal (proposal_key, config, store : proposal_key * config * storage): return =
-  let store = authorize_admin store in
-
   let proposal = check_if_proposal_exist (proposal_key, store) in
-  if is_voting_period_over(proposal, config.voting_period, store)
+
+  if   (sender = proposal.proposer)
+    || (sender = store.guardian && sender <> source) // Guardian cannot be equal to SOURCE
+    || is_proposal_expired_time_reached (proposal, config.proposal_expired_time)
   then
-    if   do_total_vote_meet_quorum_threshold(proposal, store, config.quorum_threshold)
-      && proposal.upvotes > proposal.downvotes
-    then
-      let store = unfreeze_proposer_and_voter_token
-            (config.rejected_proposal_return_value, true, proposal, config.voting_period, config.fixed_proposal_fee_in_token, store) in
-      let store = delete_proposal (proposal.start_date, proposal_key, store) in
-      (([] : operation list), store)
-    else
-      (failwith("FAIL_DROP_PROPOSAL_NOT_ACCEPTED") : return)
+    let store = unfreeze_proposer_and_voter_token
+          ( config.rejected_proposal_return_value
+          , false // Expired proposal is treated as rejected proposal regardless of its actual votes
```

Comment:
The comment is correct, however that's _only_ for expired proposals and here we may be dealing also with proposals that are dropped by their `proposer` or by the `guardian` but haven't expired yet, so I think this should be:
```suggestion
          , not proposal_is_expired // Expired proposal is treated as rejected proposal regardless of its actual votes
```

Reply:
@pasqu4le Does that means if the proposers call `drop_proposal` on their proposals the token are return in full? 

This may pose a cheat where the proposer know the proposal is not gonna get accepted (thus their token got slashed), and call `drop_proposal` beforehand to avoid this. 

What do you think about this?

Reply:
Agreed, will update the acceptance criteria as well, let's keep the current implementation.